packages/typescript-estree/src/node-utils.ts

```diff
@@ -934,6 +934,26 @@ export function nodeCanBeDecorated(node: TSNode): boolean {
   return false;
 }
 
+export function isValidAssignmentTarget(node: ts.Node): boolean {
+  switch (node.kind) {
+    case SyntaxKind.Identifier:
+      return true;
+    case SyntaxKind.PropertyAccessExpression:
+    case SyntaxKind.ElementAccessExpression:
+      if (node.flags & ts.NodeFlags.OptionalChain) {
+        return false;
+      }
+      return true;
+    case SyntaxKind.ParenthesizedExpression:
+    case SyntaxKind.TypeAssertionExpression:
```

Comment:
I might have missed it but I think we're missing a case for this

Reply:
@bradzacher Wow, thanks for catching that! In doing so I realized `TypeAssertionExpression` is only for `<number>foo` and `foo as number` is a separate syntax kind.